src/dsp_tools/xmllib/models/values.py

```diff
@@ -386,10 +386,10 @@ def make_element(self) -> etree._Element:
         attribs = {"encoding": "xml"}
         if self.permissions != Permissions.PROJECT_SPECIFIC_PERMISSIONS:
             attribs["permissions"] = self.permissions.value
-        if self.comment:
-            attribs["comment"] = self.comment
+        if is_string_like(self.comment):
+            attribs["comment"] = str(self.comment)
         ele = etree.Element(f"{DASCH_SCHEMA}text", attrib=attribs, nsmap=XML_NAMESPACE_MAP)
-        ele.text = self.value
+        ele.text = str(self.value)
```

Comment:
Since we only warn if the input is invalid it is possible to have values here that would cause a crash during the serialisation. It would also be possible in the `__post_init__`, but I thought I would not change too much... What do you think is better?
If we decide to do it in the `__post_init__` then I would check the comment there also and replace it with `None` if it is empty.

Reply:
I think crashing during serialization is not a good thing. It is then difficult for the user to understand where the error stems from. I prefer checking this in the `__post_init__`.

Reply:
So I think that `ele.text = str(self.value)` is actually a bad idea, because the user would end up with `<text>nan</text>`, without getting an error.

Reply:
They would get an error in the `__post_init__` just not during serialisation.